SPECS/k3s/k3s-1.24.6-cve.patch

```diff
@@ -0,0 +1,74 @@
+diff --git a/go.mod b/go.mod
+index ae51a2598d..c402fe477b 100644
+--- a/go.mod
++++ b/go.mod
+@@ -7,7 +7,7 @@ replace (
+ 	github.com/Mirantis/cri-dockerd => github.com/k3s-io/cri-dockerd v0.2.4-0.20220729204449-bcbb75d5abff // k3s-master
+ 	github.com/cloudnativelabs/kube-router => github.com/k3s-io/kube-router v1.5.1-0.20220630214451-a43bcd8511d2
+ 	github.com/containerd/cgroups => github.com/containerd/cgroups v1.0.1
+-	github.com/containerd/containerd => github.com/k3s-io/containerd v1.5.13-k3s1 // k3s-release/1.5
++	github.com/containerd/containerd => github.com/guptavinayak11/containerd v1.5.14-0.20230311100541-d1bb18d6c070 // k3s-release/1.5
```

Comment:
Are we sure we want to do this ?

I would request to tag someone within the `k3s` stakeholder team also please take a look and vet this :)

Reply:
Yes, so creating the fork is the only way to apply patches on the go.mod containerd dependency so that go mod vendor can package it correctly and also the go build command can build it. Is there any other way you feel we can do this?

Reply:
The go.mod container dependency is a red herring; the version mentioned in here is not at all relevant to the one used in the build, and changing the version mentioned in the go.mod won't fix the CVE as a result. 

https://github.com/k3s-io/k3s/blob/26e9405767263a2915723cb72b1ffd7f50687a8f/scripts/version.sh#L25-L27
The build process does some monkeypatching of containerd to avoid breaking dependencies. This means they use a forked version of containerd which lags behind the upstream quite a bit. You can see the monkey patching process in `scripts/download`. Their forked version does not currently have a fix for this CVE.

They are planning on fixing this in the [next patch version](https://github.com/k3s-io/k3s/issues/7092) of k3s. For now, I think we should either wait for this fix to be deployed, or apply a patch to the vendored tarball to match what their fix does.